server/neural_network_interface.py

```diff
@@ -80,42 +34,84 @@ def generate_move(self, sid, gesture_data, message_data):
                 movement_ct)
 
 class NeuralNetworkBlocksworldModel(BlocksworldModel):
-    def __init__(self, h5_path):
-        self.model = load_model(h5_path)
-
-    def convert_state(self, input_state):
-        def convert_block(block):
-            return [
-                block['blockId'][6:],
-                block['topLetter'],
-                block['topColor'],
-                block['bottomLetter'],
-                block['bottomColor'],
-                0,
-                0
-            ]
-        block_lists = chain.from_iterable(map(convert_block, input_state))
-        return np.array([list(block_lists)])
-
+    def __init__(self, h5_paths):
+        (self.flip_model, self.colors_model, self.letters_model) = runner.load_models(h5_paths)
+        self.tokenizer = core.build_tokenizer(core.load_vocabulary())
+        super().__init__()
 
     def generate_move(self, sid, gesture_data, message_data):
-        movement_ct = _movement_count_dict.setdefault(sid, 0) + 1
+        movement_ct = self._movement_count_dict.setdefault(sid, 0) + 1
+        self._movement_count_dict[sid] = movement_ct
+
         message = message_data['text']
         game_state = message_data['gameState']
 
-        state_input = encode_states(self.convert_state(game_state))
-        words_input = tokenize(message)
+        (flip, color, letter) = self._run_models(message)
+
+        block_id = _find_block(game_state, color, letter)
+
+        if not block_id:
+            print(str(flip))
+            print(str(color))
+            print(str(letter))
```

Comment:
Did you mean to leave the prints?

Reply:
Yes, though we can take them out (or move them to a log file or something). I thought we could leave them in since they show what the AI was thinking when it failed.